exercicios/para-casa/entregas/elisCoimbra/resolucao.js

```diff
@@ -0,0 +1,73 @@
+/*
+Exercicio Proposto: Considere um objeto que represente uma conta bancária, a conta possui o nome do titular, 
+o saldo e o limite da conta. É possível fazer operações de consulta de saldo, saque e depósito. No caso de saque é 
+necessário verificar se existe saldo suficiente para retirada, caso o cliente não tenha saldo suficiente para a operação,
+ mas possua limite disponível (e suficiente), o saque poderá ocorrer, nesses casos o saldo do cliente ficará negativo após o 
+ saque. Além disso, o limite de uma conta pode ser reajustado (para mais e para menos) ou desativado. Use a abordagem Red - Green
+  - Refactor para desenvolver essa aplicação.
+*/
+
+class Conta{
+    
+    constructor(nome, saldo, limite, limiteAtivado){
+        this.nome = nome;
+        this.saldo = saldo;
+        this.limite = limite;
+        this.limiteAtivado = limiteAtivado;
+    }
+
+    consultarSaldo(){
+        return this.saldo;
+    }
+    consultarLimite(){
+        return this.limite;
+    }
+    fazerDeposito(quantidade){
+        this.saldo = this.saldo+quantidade
+        return this.consultarSaldo()
+    }
+    fazerSaque(quantidade){
+        if(quantidade <= this.saldo){
+            console.log('O seu saldo é suficiente')
+            this.saldo = this.saldo - quantidade 
```

Comment:
Sugestão: nessa linha "this.saldo = this.saldo - quantidade", você pode fazer dessa forma "this.saldo -= quantidade" é uma forma mais concisa de fazer a mesma coisa.